src/main/java/org/jabref/gui/desktop/JabRefDesktop.java

```diff
@@ -83,12 +87,22 @@ public static void openExternalViewer(BibDatabaseContext databaseContext,
             openDoi(link);
             return;
         } else if (StandardField.EPRINT.equals(field)) {
-            link = ArXivIdentifier.parse(link)
-                                  .map(ArXivIdentifier::getExternalURI)
-                                  .filter(Optional::isPresent)
-                                  .map(Optional::get)
-                                  .map(URI::toASCIIString)
-                                  .orElse(link);
+            IdentifierParser identifierParser = new IdentifierParser(entry);
+            link = identifierParser.parse(StandardField.EPRINT)
+                    .map(Identifier::getExternalURI)
+                    .filter(Optional::isPresent)
```

Comment:
I think the inner filter + map check is not necessary, it will automatically go to the orElse